rest/server_context.go

```diff
@@ -484,11 +484,9 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 
 		for scopeName, scopeConfig := range config.Scopes {
 			for collectionName, _ := range scopeConfig.Collections {
-				dataStore := bucket.NamedDataStore(base.ScopeAndCollectionName{Scope: scopeName, Collection: collectionName})
-
-				// Check if scope/collection specified exists. Will enter retry loop if connection unsuccessful
-				if err := base.WaitUntilDataStoreExists(dataStore); err != nil {
-					return nil, fmt.Errorf("attempting to create/update database with a scope/collection that is not found")
+				dataStore, err := bucket.NamedDataStore(base.ScopeAndCollectionName{Scope: scopeName, Collection: collectionName})
+				if err != nil {
+					return nil, err
 				}
```

Comment:
I'm not convinced we should be pushing the `WaitUntilDataStoreExists` logic down into the Collection constructor for a couple of reasons:

- It changes the sg-bucket API to be more clunky, by now returning an error
- It makes accessing a collection a _lot_ more costly (it's actually doing a KV op under the hood for each invocation now for the exists check, in addition to the agent op to fetch kvCollectionID)

Reply:
We could cache this information on `Bucket` but there are two problems with this approach:

1. User creates collection `foo` and it comes back with CollectionID 2
2. User drops collection `foo` while SG is running but database is offline and it has CollectionID 3

So we don't know when these get invalidated. If we always get a new bucket connection when we re-create the dbcontext, or take dbContext on and offline, we could cache with a call when we connect to bucket by moving `base.Collection.GetCollectionManifest` to `Gocbv2Bucket` and store all manifests on Bucket, if we are not worried about invalidation problem. If we don't find the collectionID in this manifest (newly created collection), we could get the manifest again, and this makes getting `NamedDataStore` faster again for 99% of cases, but it could still return an error if KV is no available.

We'd store a map of _all_ collections on a bucket even though we don't use them all, which is a probably a small memory cost.